src/riak_core_security.erl

```diff
@@ -82,6 +98,28 @@
 -type bucket() :: {binary(), binary()} | binary().
 -type permission() :: {string()} | {string(), bucket()}.
 -type userlist() :: all | [string()].
+-type username() :: string().
+-type metadata_key() :: string().
+-type metadata_value() :: term().
+-type options() :: [{metadata_key(), metadata_value()}].
+
+-spec find_by_metadata(metadata_key(), metadata_value()) -> [{username(), options()}].
+find_by_metadata(Key, Value) ->
```

Comment:
This is specifically `find_user_by_metadata` - we either need to make it generic (pass in a `type` for users/groups/grants and maybe a matching function?) or rename this to `find_user_by_metadata` - I think the latter is probably appropriate as the former will make this function fairly complex.